homeassistant/components/miele/vacuum.py

```diff
@@ -0,0 +1,237 @@
+
+    on_value: int
+
+
+@dataclass
+class MieleVacuumDefinition:
+    """Class for defining vacuum entities."""
+
+    types: tuple[MieleAppliance, ...]
+    description: MieleVacuumDescription
+
+
+VACUUM_TYPES: Final[tuple[MieleVacuumDefinition, ...]] = (
+    MieleVacuumDefinition(
+        types=(MieleAppliance.ROBOT_VACUUM_CLEANER,),
+        description=MieleVacuumDescription(
+            key="vacuum",
+            on_value=14,
+            translation_key="vacuum",
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MieleConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the vacuum platform."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        MieleVacuum(coordinator, device_id, definition.description)
+        for device_id, device in coordinator.data.devices.items()
+        for definition in VACUUM_TYPES
+        if device.device_type in definition.types
+    )
+
+
+VACUUM_PHASE_TO_ACTIVITY = {
+    MieleVacuumStateCode.idle: VacuumActivity.IDLE,
+    MieleVacuumStateCode.docked: VacuumActivity.DOCKED,
+    MieleVacuumStateCode.cleaning: VacuumActivity.CLEANING,
+    MieleVacuumStateCode.going_to_target_area: VacuumActivity.CLEANING,
+    MieleVacuumStateCode.returning: VacuumActivity.RETURNING,
+    MieleVacuumStateCode.wheel_lifted: VacuumActivity.ERROR,
+    MieleVacuumStateCode.dirty_sensors: VacuumActivity.ERROR,
+    MieleVacuumStateCode.dust_box_missing: VacuumActivity.ERROR,
+    MieleVacuumStateCode.blocked_drive_wheels: VacuumActivity.ERROR,
+    MieleVacuumStateCode.blocked_brushes: VacuumActivity.ERROR,
+    MieleVacuumStateCode.check_dust_box_and_filter: VacuumActivity.ERROR,
+    MieleVacuumStateCode.internal_fault_reboot: VacuumActivity.ERROR,
+    MieleVacuumStateCode.blocked_front_wheel: VacuumActivity.ERROR,
+    MieleVacuumStateCode.paused: VacuumActivity.PAUSED,
+    MieleVacuumStateCode.remote_controlled: VacuumActivity.PAUSED,
+}
+
+
+class MieleVacuum(MieleEntity, StateVacuumEntity):
+    """Representation of a Vacuum entity."""
+
+    entity_description: MieleVacuumDescription
+    _attr_supported_features = SUPPORTED_FEATURES
+    _attr_fan_speed_list = [fan_speed.name for fan_speed in FanSpeed]
+    _attr_name = None
+
+    @property
+    def activity(self) -> VacuumActivity | None:
+        """Return activity."""
+        return VACUUM_PHASE_TO_ACTIVITY.get(
+            MieleVacuumStateCode(self.device.state_program_phase)
+        )
+
+    @property
+    def status(self) -> str:
+        """Map status text."""
+        return MieleVacuumStateCode(self.device.state_program_phase).name
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes of the vacuum cleaner."""
+        data: dict[str, Any] = {}
+        if self.status is not None:
+            data[ATTR_STATUS] = self.status
+        return data
```

Comment:
Why don't we have a separate entity for this?

Reply:
When I created this for the CC a couple of years ago, the attribute was necessary to have all fields in the vacuum card populated properly. I looked around at other vacuum implementations and I think we can delete this now. I dont have access to a Miele robot, but it is easy to revert if the community complains.